awscli/customizations/s3/fileinfobuilder.py

```diff
@@ -18,13 +20,12 @@ class FileInfoBuilder(object):
     This class takes a ``FileBase`` object's attributes and generates
     a ``FileInfo`` object so that the operation can be performed.
     """
-    def __init__(self, service, endpoint, source_endpoint=None,
+    def __init__(self, client, source_client=None,
                  parameters = None, is_stream=False):
-        self._service = service
-        self._endpoint = endpoint
-        self._source_endpoint = endpoint
-        if source_endpoint:
-            self._source_endpoint = source_endpoint
+        self._client = client
+        self._source_client = copy.copy(client)
```

Comment:
Why is this necessary?  This just makes a shallow copy of the client's `__dict__` right?  Does `self._source_client = client` not work for this scenario?